osu.Game.Tournament/Screens/Gameplay/GameplayScreen.cs

```diff
@@ -36,18 +41,41 @@ public class GameplayScreen : BeatmapInfoScreen, IProvideVideo
 
         private Drawable chroma;
 
+        private TourneyVideo fallbackGameplayVideo;
+        private readonly List<(string, TourneyVideo)> modSpecificGameplayVideos = new List<(string, TourneyVideo)>();
+
         [BackgroundDependencyLoader]
-        private void load(LadderInfo ladder, MatchIPCInfo ipc, Storage storage)
+        private void load(LadderInfo ladder, MatchIPCInfo ipc, TournamentStorage storage)
         {
             this.ipc = ipc;
 
+            var extraGameplayVideos = new List<TourneyVideo>();
+
+            // look for all video files that start with "gameplay-"
+            foreach (var name in storage.GetFiles("videos", "gameplay-*").Select(x1 => Path.GetFileNameWithoutExtension(x1[("videos".Length + 1)..])))
```

Comment:
why is `videos` being stripped like that here? pretty sure you can pass a full path to `GetFileNameWithoutExtension()` and it'll strip out any directory names anyhow and only keep the filename:

```
> System.IO.Path.GetFileNameWithoutExtension("/test/tmp.txt")
"tmp"
```

also the lambda needs a better param name than `x1`

Reply:
uh ye not sure where that x1 came from tbh... I'll fix that.

As for the videos being stripped? Uh, thats what I did first, then realized I needed to strip away file ext, and did `Path.GetFileNameWithoutExtension` and then didnt really think much more of it. I'll take a look at that as well.

Small sidenote, I started by getting an instance of `TournamentVideoResourceStore` and then calling `videoResourceStore.GetAvailableResources()` but somehow that thing is always empty, so I resorted to using normal storage. If that would have worked, then I wouldnt have needed to strip away video in the first place :P